### PR TITLE
Cleanup, 100% test coverage, add documentation and readthedocs integration

### LINKED ISSUE #1: Clean up help()
help() is currently missing all of the useful functions and displays some symbols that shouldn't be there.

### LINKED ISSUE #2: Improve code coverage to 100%
See `test_solar.py` for a list of functions that are not currently tested.

### LINKED ISSUE #3: Publish interface on readthedocs.io
The code interface should be published on readthedocs.io

### LINKED ISSUE #4: Improve README.md
Need to write a real README file.

### PR DESCRIPTION
**Pull request by rfrenchseti**
# Fixed Issues

- Fixes #1 
- Fixes #2
- Fixes #3
- Fixes #4 

# Summary of Changes

- Increase test coverage to 100%.
- Clean up the code to be (mostly) PEP8-compliant and add `.flake8` file to eliminate the two remaining errors.
- Update the automated tests to run `flake8`.
- Write a new `README.md` including installation instructions and examples.
- Convert docstrings to Google style.
- Add Sphinx documentation and `readthedocs.io` integration.
- Add `pull_request_template.md`.

# Known Problems

- None.

### CODE REVIEW

File: solar/__init__.py [hunk 1]
```diff
@@ -38,68 +53,72 @@
 # First UNIT_DICT item is conversion factor from W/m^2/um or from W/m^2/Hz.
 # Second item is True if the units are per wavelength, False if per frequency.
 UNIT_DICT = {
-    'W/m^2/um'     : (1.   , True),     # default units
-    'W/m^2/nm'     : (1.e-3, True),
-    'W/m^2/A'      : (1.e-4, True),
+    'W/m^2/um':      (1.,    True),     # default units
```
Comment (markshowalter) [inline solar/__init__.py]: Was it really necessary to dis-align my source code?

Somewhere, we should probably explain that "u" as a prefix means "micro" as in Greek mu. So "um" is microns and "uJy" is microJanskys. I guess this should really be in any docstring describing units.

File: docs/conf.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+# -- Project information -----------------------------------------------------
+# https://www.sphinx-doc.org/en/master/usage/configuration.html#project-information
+
+import os
+import sys
+
+sys.path.insert(0, os.path.abspath('..'))
+
+project = 'solar'
+copyright = '2024, Mark Showalter'
```
Comment (markshowalter) [inline docs/conf.py]: What does this copyright mean? I suppose it doesn't affect licensing. I'm not sure I want my name in lots of software, so would it make sense to use "PDS Ring-Moon Systems Node" in its place?

**Review by markshowalter — CHANGES_REQUESTED** [review]

File: solar/__init__.py (hunk 1 above)
Comment (rfrenchseti) [reply solar/__init__.py]: I have re-aligned the colons and placed an explanation for "u" in each docstring unit/xunit argument.

File: docs/conf.py (hunk 2 above)
Comment (rfrenchseti) [reply docs/conf.py]: Fixed

**Comment by rfrenchseti** [comment]
I removed the "Defaults to" sentence from each docstring argument and moved "Note:" to after the return argument.

**Pull request merged**